File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/KnowledgeBasesApi.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class KnowledgeBasesApi {
    private final HttpClient client;
    
    public KnowledgeBasesApi(HttpClient client) {
        this.client = client;
    }

    /** List documents */
    public ListDocumentsResponse listDocuments(String knowledgeBaseId, Map<String, Object> params) throws Exception {
        return (ListDocumentsResponse) client.get(ApiPaths.aiPath("/knowledge-bases/" + knowledgeBaseId + "/documents"), params);
    }

    /** Add document */
    public DocumentResponse addDocument(String knowledgeBaseId, DocumentAddRequest body) throws Exception {
        return (DocumentResponse) client.post(ApiPaths.aiPath("/knowledge-bases/" + knowledgeBaseId + "/documents"), body);
    }

    /** Search knowledge */
    public KnowledgeSearchResponse searchKnowledge(KnowledgeSearchRequest body) throws Exception {
        return (KnowledgeSearchResponse) client.post(ApiPaths.aiPath("/knowledge-bases/search"), body);
    }

    /** List knowledge bases */
    public KnowledgeBaseListResponse listKnowledgeBases(Map<String, Object> params) throws Exception {
        return (KnowledgeBaseListResponse) client.get(ApiPaths.aiPath("/knowledge-bases"), params);
    }

    /** Create knowledge base */
    public KnowledgeBaseResponse createKnowledgeBase(KnowledgeBaseCreateRequest body) throws Exception {
        return (KnowledgeBaseResponse) client.post(ApiPaths.aiPath("/knowledge-bases"), body);
    }

    /** Get document */
    public DocumentResponse getDocument(String knowledgeBaseId, String documentId) throws Exception {
        return (DocumentResponse) client.get(ApiPaths.aiPath("/knowledge-bases/" + knowledgeBaseId + "/documents/" + documentId + ""));
    }

    /** Delete document */
    public Void deleteDocument(String knowledgeBaseId, String documentId) throws Exception {
        client.delete(ApiPaths.aiPath("/knowledge-bases/" + knowledgeBaseId + "/documents/" + documentId + ""));
        return null;
    }

    /** Get knowledge base */
    public KnowledgeBaseResponse getKnowledgeBase(String knowledgeBaseId) throws Exception {
        return (KnowledgeBaseResponse) client.get(ApiPaths.aiPath("/knowledge-bases/" + knowledgeBaseId + ""));
    }

    /** Delete knowledge base */
    public Void deleteKnowledgeBase(String knowledgeBaseId) throws Exception {
        client.delete(ApiPaths.aiPath("/knowledge-bases/" + knowledgeBaseId + ""));
        return null;
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/KnowledgeBasesApi.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class KnowledgeBasesApi(private val client: HttpClient) {

    /** List documents */
    suspend fun listDocuments(knowledgeBaseId: String, params: Map<String, Any>? = null): ListDocumentsResponse? {
        return client.get(ApiPaths.aiPath("/v1/knowledge-bases/$knowledgeBaseId/documents"), params) as? ListDocumentsResponse
    }

    /** Add document */
    suspend fun addDocument(knowledgeBaseId: String, body: DocumentAddRequest): DocumentResponse? {
        return client.post(ApiPaths.aiPath("/v1/knowledge-bases/$knowledgeBaseId/documents"), body) as? DocumentResponse
    }

    /** Search knowledge */
    suspend fun searchKnowledge(body: KnowledgeSearchRequest): KnowledgeSearchResponse? {
        return client.post(ApiPaths.aiPath("/v1/knowledge-bases/search"), body) as? KnowledgeSearchResponse
    }

    /** List knowledge bases */
    suspend fun listKnowledgeBases(params: Map<String, Any>? = null): KnowledgeBaseListResponse? {
        return client.get(ApiPaths.aiPath("/v1/knowledge-bases"), params) as? KnowledgeBaseListResponse
    }

    /** Create knowledge base */
    suspend fun createKnowledgeBase(body: KnowledgeBaseCreateRequest): KnowledgeBaseResponse? {
        return client.post(ApiPaths.aiPath("/v1/knowledge-bases"), body) as? KnowledgeBaseResponse
    }

    /** Get document */
    suspend fun getDocument(knowledgeBaseId: String, documentId: String): DocumentResponse? {
        return client.get(ApiPaths.aiPath("/v1/knowledge-bases/$knowledgeBaseId/documents/$documentId")) as? DocumentResponse
    }

    /** Delete document */
    suspend fun deleteDocument(knowledgeBaseId: String, documentId: String): Unit {
        client.delete(ApiPaths.aiPath("/v1/knowledge-bases/$knowledgeBaseId/documents/$documentId"))
    }

    /** Get knowledge base */
    suspend fun getKnowledgeBase(knowledgeBaseId: String): KnowledgeBaseResponse? {
        return client.get(ApiPaths.aiPath("/v1/knowledge-bases/$knowledgeBaseId")) as? KnowledgeBaseResponse
    }

    /** Delete knowledge base */
    suspend fun deleteKnowledgeBase(knowledgeBaseId: String): Unit {
        client.delete(ApiPaths.aiPath("/v1/knowledge-bases/$knowledgeBaseId"))
    }
}
